helpers/calendar/analytics_helper.py

```diff
@@ -224,3 +226,34 @@ def get_list_of_month_dates(start_date, start_dt, end_date, end_dt):
         last_month_start_date = datetime.strptime(last_month_start, '%b %d %Y').isoformat() + 'Z'  # noqa E501
         dates.append([last_month_start_date, end_date])
         return dates
+
+    @staticmethod
+    def get_day_month_list(start_date, end_date, number_of_days, date_pattern):
+        freq = "D" if number_of_days <= 30 else "MS"
+        return pandas.date_range(
+            start_date, pandas.Timestamp(end_date) - pandas.DateOffset(days=1),
+            freq=freq).strftime(date_pattern).tolist()
+
+    @staticmethod
+    def get_bookings_count_from_events(start_date, end_date, events, room_id):
+        first_day = dateutil.parser.parse(start_date)
+        last_day = dateutil.parser.parse(end_date)
+        number_of_days = (last_day - first_day).days
+        event_index = 0
+        date_pattern = "%b %d %Y" if number_of_days <= 30 else "%b %Y"
+        period_list = CommonAnalytics.get_day_month_list(
+            start_date, end_date, number_of_days, date_pattern)
+        bookings_with_period = {}
```

Comment:
The variable name implies a list of bookings but an assignment is an object. Why is this so?

Reply:
It is a dictionary that maps a period to its bookings. I have renamed it to `period_to_bookings`